core/src/main/java/com/alexafit/core/local/datastore/DataStorePreferenceStorageImpl.kt

```diff
@@ -0,0 +1,24 @@
+package com.alexafit.core.local.datastore
+
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class DataStorePreferenceStorageImpl @Inject constructor(
+    private val dataStore: DataStore<Preferences>
+) : PreferenceStorage {
+
+    override suspend fun setAuthorizationKey(authorizationKey: String?) {
+        dataStore.edit { settings ->
+            settings[PreferenceStorage.PreferencesKey.PREF_AUTHORIZATION_KEY] = authorizationKey ?: ""
+        }
+    }
+
+    override val authorizationKey: Flow<String> =
+        dataStore.data.map { it[PreferenceStorage.PreferencesKey.PREF_AUTHORIZATION_KEY] ?: "" }
```

Comment:
When setting an authorization key, it's optional (because it can be null). When reading it, it's suddenly not optional anymore and will always give you a value. I'd recommend to keep return null in order to indicate a value doesn't exist. Otherwise, it might be confusing since `Flow<String>` sounds like there is always a key that's returned which is not true, since it can be empty